src/arrayMethodSort.js

```diff
@@ -3,9 +3,28 @@
 /**
  * Implement method Sort
  */
+
+function compareString(a, b) {
+  if (String(a) > String(b)) {
+    return 1;
+  } else if (String(a) < String(b)) {
+    return -1;
+  } else {
+    return 0;
+  }
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareString) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction(this[i], this[j]) > 0) {
+          [this[i], this[j]] = [this[j], this[i]];
+        }
+      }
+    }
```

Comment:
let's use `do while` instead of nested loops

Reply:
updated